### PR TITLE
Enhance Wallet Balance Calculation Tests with Parameterized Cases

### LINKED ISSUE #308: Implement Wallet Balance Tracking
User story link: bycza-zagroda/story#23

As a user of the TrackExpenseApp I want to view the current balance for each of my wallets in order to have a clear understanding of my financial status for each specific wallet.

Definition of Done:

- The balance is successfully displayed for each wallet in the "MY WALLETS" section.

### PR DESCRIPTION
**Pull request by Dharakkkk**
## Summary
This pull request introduces enhanced testing for the wallet balance calculation feature in the TrackExpenseApp. By implementing parameterized tests, we now cover a broader range of scenarios, ensuring more robust and reliable balance computation.

## Key Changes
- **Parameterized Tests:** Implemented parameterized tests in `WalletTest` to validate the balance calculation logic under various scenarios.
- **Comprehensive Scenarios:** Added test cases to cover typical use cases, edge cases, and unusual transaction patterns. This includes testing with a mix of income and expense transactions, extreme transaction amounts, empty wallet scenarios, and transactions from different time periods.
- **Code Readability:** Improved the readability and structure of the test code, making it easier to understand and maintain.

## Benefits
- **Enhanced Coverage:** The new tests provide a more comprehensive assessment of the balance calculation logic, ensuring accuracy across a wide range of scenarios.
- **Maintenance and Scalability:** With parameterized tests, it's easier to add new scenarios in the future, enhancing the maintainability and scalability of our test suite.
- **Reliability:** These improvements contribute to the overall reliability and robustness of the application by ensuring that the balance calculation behaves as expected under various conditions.

## Additional Notes
- All tests have been run and pass successfully.
- These changes are confined to the test suite and do not impact the application's core functionality or user experience.

Looking forward to your feedback and suggestions for any further improvements.

#### Checklist template for LEADERS

&#x2611; | &#x2610;

&#x2611; New code is covered with tests and coverage requirements are met.
&#x2611; Build is not failing.
&#x2611; All tests are passing.
&#x2611; All conflicts are resolved, and the branch is rebased on top of the up-to-date parent branch.
&#x2610; All issue requirements are implemented.
&#x2611; At least two reviewers are assigned.
&#x2611; The Pull request is linked to the related issue.
&#x2610; Correct branch name. Convention [https://github.com/bycza-zagroda/track-expenses-app-documentation/blob/develop/githubwork/GITHUB_WORK.md]
&#x2611; Correct unit test name - suffix 'Test' in the end.
&#x2611; Correct integration test name - suffix 'IT' in the end.
&#x2611; Positive and negative test paths have been implemented.

### CODE REVIEW

**Review by dawciobiel — CHANGES_REQUESTED** [review]
The code has been implemented.
The test has as well.

However, for now the method:
```java
    BigDecimal calculateCurrentBalance(List<FinancialTransactionDTO> transactions);
```

is not called anywhere. It should be called at least at:
```java
    WalletDTO findById(@Min(1) @NotNull Long walletId, Long userId);
```

And preferably at:
```java
    List<WalletDTO> getWallets(Long userId);

    WalletDTO findById(@Min(1) @NotNull Long walletId, Long userId);

    List<WalletDTO> findAllByNameIgnoreCase(@NotBlank @Length(max = 20) @Pattern(regexp = "[}]+") String name, Long userId);
```

By the method:
```java
    WalletDTO updateWallet(@Min(1) @NotNull Long id, @Valid WalletUpdateDTO walletToUpdate, Long userId);
```

As a balance value, you can necessarily either call `calculateCurrentBalance()` or insert as a NULL value.
This is because if this is not done, the value 0 will be inserted. And this would not be true.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/wallet/impl/WalletServiceImpl.java [hunk 1]
```diff
@@ -64,13 +73,29 @@ public WalletDTO updateWallet(@Min(1) @NotNull Long id, @Valid WalletUpdateDTO d
...
 
-        return walletModelMapper.mapWalletEntityToWalletDTO(wallet);
+        List<FinancialTransaction> transactions = transactionRepository.findAllByWalletIdAndWalletUserIdOrderByDateDesc(id, userId);
+        List<FinancialTransactionDTO> transactionDTOs = transactions.stream()
+                .map(transactionModelMapper::mapFinancialTransactionEntityToFinancialTransactionDTO)
+                .collect(Collectors.toList());
+        BigDecimal balance = calculateCurrentBalance(transactionDTOs);
+
+        return new WalletDTO(wallet.getId(), wallet.getName(), wallet.getCreationDate(), wallet.getUser().getId(), balance);
+
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/wallet/impl/WalletServiceImpl.java]: Empty line not needed

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/wallet/impl/WalletServiceImpl.java [hunk 2]
```diff
@@ -36,6 +43,8 @@ public class WalletServiceImpl implements WalletService {
     private final WalletRepository walletRepository;
     private final WalletModelMapper walletModelMapper;
     private final AuthRepository userRepository;
+    private final FinancialTransactionRepository transactionRepository;
+    private final FinancialTransactionModelMapper transactionModelMapper;
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/wallet/impl/WalletServiceImpl.java:52]: Please insert blank lines between class fields

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/wallet/impl/WalletServiceImpl.java [hunk 3]
```diff
@@ -64,13 +73,29 @@ public WalletDTO updateWallet(@Min(1) @NotNull Long id, @Valid WalletUpdateDTO d
...
         return walletRepository.findAllByUserIdOrderByNameAsc(userId).stream()
-                .map(walletModelMapper::mapWalletEntityToWalletDTO)
+                .map(wallet -> {
+                    List<FinancialTransaction> transactions = transactionRepository
+                            .findAllByWalletIdOrderByDateDesc(wallet.getId());
+                    List<FinancialTransactionDTO> transactionDTOs = transactions.stream()
+                            .map(transactionModelMapper::mapFinancialTransactionEntityToFinancialTransactionDTO)
+                            .collect(Collectors.toList());
+                    BigDecimal balance = calculateCurrentBalance(transactionDTOs);
+                    return new WalletDTO(wallet.getId(), wallet.getName(),
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/wallet/impl/WalletServiceImpl.java:101]: Please insert one blank line before return statement

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/wallet/impl/WalletServiceImpl.java [hunk 4]
```diff
@@ -93,17 +118,25 @@ public void deleteWalletById(@Min(1) @NotNull Long walletId, Long userId) {
...
             );
         }
-        return walletModelMapper.mapWalletEntityToWalletDTO(wallet);
+        List<FinancialTransaction> transactions = transactionRepository
+                .findAllByWalletIdAndWalletUserIdOrderByDateDesc(walletId, userId);
+        List<FinancialTransactionDTO> transactionDTOs = transactions.stream()
+                .map(transactionModelMapper::mapFinancialTransactionEntityToFinancialTransactionDTO)
+                .collect(Collectors.toList());
+        BigDecimal balance = calculateCurrentBalance(transactionDTOs);
+        return new WalletDTO(wallet.getId(), wallet.getName(),
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/wallet/impl/WalletServiceImpl.java:141]: Please insert one blank line before return statement

**Review by dawciobiel — DISMISSED** [review]
LGTM

**Review by dawciobiel — CHANGES_REQUESTED** [review]
LGTM but improved code formatting is required

**Review by dawciobiel — APPROVED** [review]
LGTM
